packages/govuk-frontend/src/govuk/components/button/button.test.js

```diff
@@ -55,105 +55,121 @@ describe('/components/button', () => {
   })
 
   describe('preventing double clicks', () => {
-    // Click counting is done through using the button to submit
-    // a form and counting submissions. It requires some bits of recurring
-    // logic which are wrapped in the following helpers
-
     /**
-     * Wraps the button rendered on the page in a form
-     *
-     * Examples don't do this and we need it to have something to submit
+     * Sets the number of times a button was clicked
      *
-     * @param {import('puppeteer').Page} page - Puppeteer page object
+     * @param {import('puppeteer').ElementHandle<HTMLButtonElement>} $button - Puppeteer Button element
      * @returns {Promise<void>}
      */
-    function trackClicks (page) {
-      return page.evaluate(() => {
-        const $button = document.querySelector('button')
-        const $form = document.createElement('form')
-        $button.parentNode.appendChild($form)
-        $form.appendChild($button)
-
-        globalThis.__SUBMIT_EVENTS = 0
-        $form.addEventListener('submit', (event) => {
-          globalThis.__SUBMIT_EVENTS++
-          // Don't refresh the page, which will destroy the context to test against.
-          event.preventDefault()
-        })
-      })
+    async function trackClicks ($button) {
+      const counts = {
+        click: 0,
+        debounce: 0
+      }
+
+      // Track number of button clicks
+      return $button.evaluate((el, counts) => el.addEventListener('click', (event) => {
+        counts.click++
+        el.dataset.clickCount = `${counts.click}`
+
+        // Track number of button clicks that debounced
+        event.preventDefault = () => {
```

Comment:
Replaces `event.preventDefault()` with a stub that counts every time we debounce a click